apps/geo/serializers.py

```diff
@@ -148,3 +153,38 @@ class Meta:
             'region_title', 'admin_level_level', 'admin_level_title',
             'parent'
         )
+
+
+class RegionGqSerializer(ProjectPropertySerializerMixin, UserResourceSerializer):
+    project = IntegerIDField()
+
+    class Meta:
+        model = Region
+        exclude = ('geo_options', 'is_published')
+
+    def validate_project(self, project):
+        try:
+            project = Project.objects.get(id=project)
+        except Project.DoesNotExist:
+            raise serializers.ValidationError(
+                'Project matching query does not exist'
+            )
+
+        if not project.can_modify(self.context['request'].user):
+            raise serializers.ValidationError('Permission Denied')
+        return project.id
+
+    def validate(self, data):
+        if self.instance and self.instance.is_published:
+            raise serializers.ValidationError('Published region can\'t be changed. Please contact Admin')
+        return data
+
+    def create(self, validated_data):
+        project = validated_data.pop('project', None)
+        region = super().create(validated_data)
+
+        if project:
+            project = Project.objects.get(id=project)
```

Comment:
Project will be an object here. If we return project as object from validate_project.

```suggestion
        if project:
```